### PR TITLE
Address reviewer feedback

### PR DESCRIPTION
**Pull request by rbuckton**
Addresses some reviewer feedback from #10.

### CODE REVIEW

File: spec/sec-regexp-regular-expression-objects-patch.html [hunk 1]
```diff
@@ -280,71 +280,50 @@ <h1><ins>GetStringIndex ( _S_, _Input_, _e_ )</ins></h1>
...
     <emu-clause id="sec-getmatchresult" aoid="GetMatchResult">
-      <h1><ins>GetMatchResult ( _Input_, _startIndex_, _endIndex_, _fullUnicode_ _matchOptions_ )</ins></h1>
-      <p>The abstract operation GetMatchResult with arguments _Input_, _startIndex_, _endIndex_, _fullUnicode_, and _matchOptions_ performs the following steps:</p>
+      <h1><ins>GetMatchResult ( _S_, _startIndex_, _endIndex_, _matchOptions_ )</ins></h1>
+      <p>The abstract operation GetMatchResult with arguments _S_, _startIndex_, _endIndex_, and _matchOptions_ performs the following steps:</p>
       <emu-alg>
-      1. Assert: _Input_ is a List of either Unicode code points or code units.
+      1. Assert: Type(_S_) is String.
+      1. Assert: _startIndex_ is an integer index into _S_.
+      1. Assert: _endIndex_ is an integer index into _S_ that is greater than or equal to _startIndex_.
```
Comment (ljharb) [inline spec/sec-regexp-regular-expression-objects-patch.html:296]: perhaps? (not a blocker)
```suggestion
      1. Assert: _endIndex_ is an integer index into _S_, and _endIndex_ &gte; _startIndex_.
```

File: spec/sec-regexp-regular-expression-objects-patch.html [hunk 2]
```diff
@@ -280,71 +280,50 @@ <h1><ins>GetStringIndex ( _S_, _Input_, _e_ )</ins></h1>
...
         1. Return CreateArrayFromList(&laquo; _startIndex_, _endIndex_ &raquo;).
       1. Else,
-        1. Let _matchChars_ be a new List whose elements are the characters of _Input_ at indices _startIndex_ (inclusive) through _endIndex_ (exclusive).
-        1. If _fullUnicode_ is *true*, then
-          1. Assert: _matchChars_ is a List of code points.
-          1. Return the String value whose code units are the UTF16Encoding of the code points of _matchChars_.
-        1. Else _fullUnicode_ is *false*,
-          1. Assert: _matchChars_ is a List of code units.
-          1. Return the String value consisting of the code units of _matchChars_.
+        1. Return the portion of _S_ between offset _startIndex_ inclusive and offset _endIndex_ exclusive.
```
Comment (ljharb) [inline spec/sec-regexp-regular-expression-objects-patch.html:301]: i wonder if we might want to extract part of https://tc39.es/ecma262/#sec-string.prototype.slice into an abstract operation, and then we could reuse that here? (also not a blocker)

Comment (rbuckton) [reply spec/sec-regexp-regular-expression-objects-patch.html:301]: This is reusing existing spec text originally in RegExpBuiltinExec in the current version of the spec.

File: spec/sec-regexp-regular-expression-objects-patch.html (hunk 1 above)
Comment (rbuckton) [reply spec/sec-regexp-regular-expression-objects-patch.html:296]: The spec isn't consistent on this. There are more than a few cases of english wording for such comparisons.

Comment (ljharb) [reply spec/sec-regexp-regular-expression-objects-patch.html:296]: I can only find 3 occurrences of "greater than or equal to" in actual abstract operation steps (as opposed to notes); one is in RegExpBuiltinExec, and the other 2 are in `Runtime Semantics: Decode`; but I can find none using `&gte;`. I'll open a PR to clean these up.

File: spec/sec-regexp-regular-expression-objects-patch.html (hunk 2 above)
Comment (ljharb) [reply spec/sec-regexp-regular-expression-objects-patch.html:301]: RegExpBuiltInExec could use some refactoring anyways :-) fine if you'd prefer not to do this tho; it was just a thought.

File: spec/sec-regexp-regular-expression-objects-patch.html [hunk 3]
```diff
@@ -280,71 +280,50 @@ <h1><ins>GetStringIndex ( _S_, _Input_, _e_ )</ins></h1>
...
+        1. Let _end_ be ? Get(_match_, `"1"`).
+        1. If Type(_start_) is not Number, throw a *TypeError* exception.
+        1. If Type(_end_) is not Number, throw a *TypeError* exception.
+        1. Let _isEmpty_ be _start_ = _end_.
       1. Else,
-        1. Let _matchStr_ be ? ToString(_match_).
-        1. If _matchStr_ is the empty String, then
-          1. Return True.
-        1. Return False.
+        1. Set _match_ to ? ToString(_match_).
```
Comment (waldemarhorwat) [inline spec/sec-regexp-regular-expression-objects-patch.html]: Can spec algorithms mutate their parameters like this? The Algorithm Conventions section (5.2) only describes "Set _x_ to ..." applying to things defined by "Let _x_ be ..." earlier in the same algorithm, not to algorithm parameters.

**Review by waldemarhorwat — CHANGES_REQUESTED** [review]
Please address the Algorithm Conventions issue. The rest looks good.

File: spec/sec-regexp-regular-expression-objects-patch.html [hunk 4]
```diff
@@ -280,71 +280,51 @@ <h1><ins>GetStringIndex ( _S_, _Input_, _e_ )</ins></h1>
...
+        1. If Type(_start_) is not Number, throw a *TypeError* exception.
+        1. If Type(_end_) is not Number, throw a *TypeError* exception.
+        1. Let _matchValue_ be _match_.
+        1. Let _isEmpty_ be _start_ = _end_.
       1. Else,
-        1. Let _matchStr_ be ? ToString(_match_).
-        1. If _matchStr_ is the empty String, then
-          1. Return True.
-        1. Return False.
+        1. Let _matchValue_ to ? ToString(_match_).
```
Comment (ljharb) [inline spec/sec-regexp-regular-expression-objects-patch.html]:
```suggestion
        1. Let _matchValue_ be ? ToString(_match_).
```

File: spec/sec-regexp-regular-expression-objects-patch.html (hunk 1 above)
Comment (rbuckton) [reply spec/sec-regexp-regular-expression-objects-patch.html:296]: Will you be switching the spec to &gte; in those cases?

Comment (ljharb) [reply spec/sec-regexp-regular-expression-objects-patch.html:296]: I'm not sure yet; for now i'm going to resolve this - we can smooth out any inconsistency at spec PR time, if there is any by then.

**Comment by rbuckton** [comment]
@waldemarhorwat any other concerns or should I go ahead and merge?

**Review by waldemarhorwat — APPROVED** [review]

**Comment by waldemarhorwat** [comment]
Go ahead and merge.

**Pull request merged**